app/screens/edit_profile/edit_profile.tsx

```diff
@@ -0,0 +1,308 @@
+        } catch (e) {
+            return resetScreen(e as Error);
+        }
+
+        return null;
+    };
+
+    const resetScreen = (resetError: Error) => {
+        setError(resetError?.message);
+        setUpdating(false);
+        enableSaveButton(true);
+        scrollViewRef.current?.scrollToPosition(0, 0, true);
+    };
+
+    const updateField = useCallback((id: string, name: string) => {
+        const update = {...userInfo};
+        update[id] = name;
+        setUserInfo(update);
+
+        // @ts-expect-error access object property by string key
+        const currentValue = currentUser[id];
+        const didChange = currentValue !== name;
+        enableSaveButton(didChange);
+    }, [userInfo]);
+
+    const setScrollViewRef = useCallback((ref) => {
+        scrollViewRef.current = ref;
+    }, []);
+
+    const renderProfilePicture = () => {
+        return (
+            <View style={style.top}>
+                <ProfilePicture
+                    author={currentUser}
+                    size={153}
+                    showStatus={false}
+                />
+            </View>
+        );
+    };
+
+    if (updating) {
+        return <ProfileUpdating/>;
+    }
+
+    return (
+        <>
+            {isTablet &&
+                <TabletTitle
+                    action={rightButton.text}
+                    enabled={canSave}
+                    onPress={submitUser}
+                    testID='custom_status.done.button'
+                    title={intl.formatMessage({id: 'mobile.screen.your_profile', defaultMessage: 'Your Profile'})}
+                />
+            }
+            <SafeAreaView
+                edges={edges}
+                style={style.flex}
+                testID='edit_profile.screen'
+            >
+                <KeyboardAwareScrollView
+                    bounces={false}
+                    innerRef={setScrollViewRef}
+                    testID='edit_profile.scroll_view'
+                >
+                    {error && <ProfileError error={error}/>}
+                    {renderProfilePicture()}
+                    <Field
+                        id={'firstName'}
+                        isDisabled={(['ldap', 'saml'].includes(service) && lockedFirstName) || ['gitlab', 'google', 'office365'].includes(service)}
+                        label={FIELDS.firstName}
+                        onChange={updateField}
+                        testID={'edit_profile.text_setting.firstName'}
+                        value={userInfo.firstName}
+                    />
+                    <View style={style.separator}/>
+                    <Field
+                        id={'lastName'}
+                        isDisabled={(['ldap', 'saml'].includes(service) && lockedLastName) || ['gitlab', 'google', 'office365'].includes(service)}
+                        label={FIELDS.lastName}
+                        onChange={updateField}
+                        testID={'edit_profile.text_setting.lastName'}
```

Comment:
```suggestion
                        testID='edit_profile.text_setting.lastName'
```

I think you get the point by now